core/src/main/java/org/apache/accumulo/core/iterators/IteratorEnvironment.java

```diff
@@ -25,24 +25,47 @@
 import org.apache.accumulo.core.data.Value;
 import org.apache.accumulo.core.iterators.IteratorUtil.IteratorScope;
 import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.spi.common.ServiceEnvironment;
 
 public interface IteratorEnvironment {
 
-  SortedKeyValueIterator<Key,Value> reserveMapFileReader(String mapFileName) throws IOException;
+  default SortedKeyValueIterator<Key,Value> reserveMapFileReader(String mapFileName)
+      throws IOException {
+    throw new UnsupportedOperationException();
+  }
 
-  AccumuloConfiguration getConfig();
+  /**
+   * @deprecated since 2.0.0 use {@link #getServiceEnv()} and
```

Comment:
Would be nice to say why its deprecated, it returns an internal type that does not have a stable API.